ci/tests/test.py

```diff
@@ -0,0 +1,120 @@
+import argparse
+import requests
+from datetime import datetime
+import random
+import pprint
+import subprocess
+import sys
+import os
+from pprint import pprint as pp
+
+test = os.getenv("test")
+run_id = os.getenv("run_id")
+branch = os.getenv("branch")
+url = os.getenv("url")
+location = os.getenv("location")
+maintainer_email = os.getenv("contact")
+imnumber = os.getenv("imnumber")
+
+config = { "maintainer_email" : "morton30@llnl.gov"}
+
+extras = { "config" : config,
+           "Test": test,
+           "git_branch" : branch,
+           "start_time" : str(datetime.now())
+         }
+
+if imnumber:
+    config["IM Number"] = imnumber
+
+data = { "run_id" : run_id,
+         "branch": branch,
+        }
+
+def get_conf():
+    results = {}
+    data.update( {"test_name" : "Set Environment",
+                  "results" : results,
+                  "extras": extras,
+                  "test_start_time": str(datetime.now()),
+                  "test_end_time" : str(datetime.now()),
+                  'function' : '_discover_environment',
+                  "module" : '_conftest' })
+    return data
+
+
+def get_result(command, name, stdout):
+    if not name:
+        name = command.split()[0]
+    start = str(datetime.now())
+
+    try:
+        out = subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT).decode("utf-8")
+        ret = True
+    except subprocess.CalledProcessError as exc:
+        out = exc.output.decode("utf-8")
+        ret = False
+        print(f"Test: {name} failed.\n{out}")
+
+    end = str(datetime.now())
+    results = { name:
+                       {"passed" : ret},
+              }
+    data.update({ "test_name" : name,
+                  "results" : results,
+                  "test_start_time": start,
+                  "test_end_time" : end,
+                  "extras": {},
```

Comment:
Shouldn't this also use the `extras` from earlier, like the `start` and `end` data?